src/cmds/fs/ls.c

```diff
@@ -168,27 +168,40 @@ int main(int argc, char **argv) {
 	if (optind < argc) {
 		struct stat sb;
 
-		if (-1 == stat(argv[optind], &sb)) {
-			return -errno;
-		}
+		optind--;
 
-		if (!S_ISDIR(sb.st_mode)) {
-			printer(argv[optind], &sb);
-			return 0;
-		}
+		while (optind++ < argc - 1) {
+
+		    if (-1 == stat(argv[optind], &sb)) {
+			    return -errno;
+            }
+
+            if (!S_ISDIR(sb.st_mode)) {
+                printer(argv[optind], &sb);
+                continue;
+		    }
+
+            sprintf(dir_name, "%s", argv[optind]);
```

Comment:
It's not safe to do `sprintf` here, because your dir_name is only `NAME_MAX` in size, while the length of argv[optind] can be greater.

Reply:
So do I need to add a judgement？

Reply:
Sorry, I do not understand what judgement means in this context, but I was thinking about `snprintf`.

Reply:
Is you mean this ?    snprintf(dirname,  NAME_MAX, "%s", argc[optind])
Here are two options， one is telling people the length of path name or file name is too long. The other one is  to avoid making mistake but people don't realize the results match the command.  
![222](https://user-images.githubusercontent.com/40829803/75761146-8445a380-5d73-11ea-97b4-a7f02e69ac3e.png)

Reply:
I  am all right. Could you check it？Thanks for your great code review. It's my first time and I learned a lot. Thanks again.

Reply:
It's a good question about how to handle long names, my point of view here is only to make long names handling to be safe. Your current variant with strncpy is good enough.